### PR TITLE
Custom controller that starts and connects to cluster

### PR DESCRIPTION
**Pull request by leahnp**
First step towards completing #RS-108

Since we are all working on the controller, this commit is to help us start visualizing how to pull this all together. Currently this controller runs locally on a mac (see readme) and connects via kubeconfig to minikube.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+	c := &Controller{
+		clientset: kubeclientset,
+	}
+	return c
+}
+
+
+func main() {
+	// pass kubeconfig like: -kubeconfig=$HOME/.kube/config
+	kubeconf := flag.String("kubeconf", "admin.conf", "Path to a kube config. Only required if out-of-cluster.")
```
Comment (alejandroEsc) [inline main.go]: Please let's not use flags. Remove and instead change to an environment variable.

File: main.go [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+	flag.Parse()
+
+	config, err := getClientConfig(*kubeconf)
+	if err != nil {
+		panic(err.Error())
```
Comment (alejandroEsc) [inline main.go]: graceful stops. Remove panic, set an error and you can optionally use a function that logs error and exists.

File: main.go [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+	"github.com/golang/glog"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+
+	// TODO build our own cluster-controller informers and clientset
+	"k8s.io/sample-controller/pkg/signals"
+)
```
Comment (oneilcin) [inline main.go]: This should be our directory structure:
github.com/samsung-cnct/cluster-controller/pkg/signals

You might need create the pkg dir and put a template signals file in there.

Comment (oneilcin) [reply main.go]: Actually if you meant handlers, that is another directory, and we would need a sample handler in the dir.

File: Dockerfile [hunk 4]
```diff
@@ -1 +1,9 @@
-<TODO>
\ No newline at end of file
+FROM golang:1.8
```
Comment (alejandroEsc) [inline Dockerfile]: we may want to use `ubuntu:16.04` unless you mean here we are using this to build?

Comment (alejandroEsc) [reply Dockerfile]: ahh ok you did, ignore me for now. Later however, i suspect juju will have to run on this same container. We can do that on a later build if thats the case.

File: README.md [hunk 5]
```diff
@@ -1,2 +1,15 @@
 # cluster-controller
-Kubernetes custom controller for managing Kubernetes clusters
+
+This repository implements a controller for watching Cluster resources as
+defined with a CustomResourceDefinition (CRD).
+
```
[the remark this answers is not in the document]
Comment (leahnp) [reply README.md]: Added the link - I am trying to add the codegen dir with CI, hopefully coming later today

**Comment by alejandroEsc** [comment]
good first commit, lgtm, going to merge.

**Pull request merged**